### PR TITLE
Making class cleanup work again

### PR DESCRIPTION
**Pull request by amyschoen**
Deleting old student repos was no longer working with the refactor.  This fix works for me.  I pulled in some of the newer  functions and the HTTPie library.

### CODE REVIEW

File: script/clean-up-class [hunk 1]
```diff
@@ -5,44 +5,40 @@
...
-    curl -s -S -u "$TOKEN_OWNER:$TEACHER_PAT" -X GET "https://$INSTANCE_URL/repos/$CLASS_ORG/$COLLAB_REPO/collaborators?affiliation=direct&per_page=100" | jq -r 'map(.login) | unique | @sh' | tr -d \'
-  )" >>log.out 2>&1
+# shell variables
+collab_repo=$1
+org_url="https://$ROOT_URL/$CLASS_ORG"
+org_repos_endpoint="https://$INSTANCE_URL/repos/$CLASS_ORG"
 
-  # Clean up conflict repos based on collaborators in the inital class repo
-  for USER_NAME in "${COLLABS[@]}"; do
+function delete_repos() {
```
Comment (parkerbxyz) [inline script/clean-up-class:22]:
```suggestion
delete_repos() {
```

I recommend removing `function` here, as it reduces compatibility with older versions of bash.

Source: https://github.com/dylanaraps/pure-bash-bible#function-declaration

File: script/clean-up-class [hunk 2]
```diff
@@ -5,44 +5,40 @@
...
-    CleanUpRepo "github-games-$USER_NAME"
+    delete_repo "conflict-practice" "$username"
+    delete_repo "github-games" "$username"
   done
 }
 
-function CleanUpRepo() {
-  # Get the Repo
-  REPO=$1
+function delete_repo() {
```
Comment (parkerbxyz) [inline script/clean-up-class:30]:
```suggestion
delete_repo() {
```

I recommend removing `function` here, as it reduces compatibility with older versions of bash.

Source: https://github.com/dylanaraps/pure-bash-bible#function-declaration

**Review by parkerbxyz — COMMENTED** [review]
Thanks for your contributions, as always, @amyschoen! ✨

**Comment by parkerbxyz** [comment]
One other recommendation here would be to update the description on line 3 to be more clear about what this script does: https://github.com/githubtraining/training-manual/blob/aaa670aeb4496212fc9a2323375913a5fba693e7/script/clean-up-class#L3

**Comment by amyschoen** [comment]
@parkerbxyz, I did not know that function is technically less correct.  Do we want to remove it everywhere and add that check to the linter?  I just followed the same format that was elsewhere in the code.

I didn't modify the comment, but I'll get that fixed up.  I'll wait on your answer for the script changes.

**Comment by parkerbxyz** [comment]
> Do we want to remove it everywhere and add that check to the linter? I just followed the same format that was elsewhere in the code.

I’ve just been doing it as I’ve refactored specific scripts. Here are the scripts that have been refactored and no longer contain the `function` syntax:

- [ask](https://github.com/githubtraining/training-manual/blob/main/script/ask)
- [bootstrap](https://github.com/githubtraining/training-manual/blob/main/script/bootstrap)
- [create-files](https://github.com/githubtraining/training-manual/blob/main/script/create-files)
- [create-initial-repo](https://github.com/githubtraining/training-manual/blob/main/script/create-initial-repo)
- [create-practice-repos](https://github.com/githubtraining/training-manual/blob/main/script/create-practice-repos)
- [shared_functions](https://github.com/githubtraining/training-manual/blob/main/script/shared_functions)
- [teach-class](https://github.com/githubtraining/training-manual/blob/main/script/teach-class)

I wouldn’t worry about removing the syntax from the other scripts unless/until we have other changes to make to them. It has actually helped me keep track of which scripts have been refactored and which still need some attention.

File: script/clean-up-class [hunk 3]
```diff
@@ -1,48 +1,44 @@
 #!/usr/bin/env bash
 #
-# Cleanup Class
+# Cleans up student repos associated with a particular GitHub class
```
[the remark this answers is not in the document]
Comment (amyschoen) [reply script/clean-up-class]: Much better.  I agree.  Suggestion accepted.  😀

**Review by parkerbxyz — APPROVED** [review]

**Review by yosselin-babel — COMMENTED** [review]
li

**Review by yosselin-babel — COMMENTED** [review]
l

**Comment by amyschoen** [comment]
@parkerbxyz are we good to merge this one?  Someone leaving junk comments made me realize this was still active.  Looks like my permissions got bumped up because I now have the ability to merge, but I don't want to screw up any of your internal processes.

**Comment by amyschoen** [comment]
🎉 My first merge to the repo.  😀

**Pull request merged**